cmd/acra-translator/server/server.go

```diff
@@ -115,30 +108,31 @@ func (server *ReaderServer) HandleConnectionString(parentContext context.Context
 
 	errCh := make(chan error)
 
-	listenerContext := server.listenerContext(parentContext)
-
 	// start accept new connections from connectionString
-	connectionChannel, err := common.AcceptConnections(listenerContext, connectionString, errCh)
+	connectionChannel, err := common.AcceptConnections(parentContext, listener, errCh)
 	if err != nil {
 		logger.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCantAcceptNewConnections).WithError(err).Errorf("Can't start to handle connection string %v", connectionString)
 		return err
 	}
 	// use to send close packets to all unclosed connections at end
-	server.waitGroup.Add(1)
+	server.backgroundWorkersSync.Add(1)
 	go func() {
-		defer server.waitGroup.Done()
+		defer server.backgroundWorkersSync.Done()
 		logger.WithField("connection_string", connectionString).Debugln("Start wrap new connections")
 		for {
 			var connection net.Conn
 			select {
 			case connection = <-connectionChannel:
 				break
 			case <-parentContext.Done():
-				logger.WithError(parentContext.Err()).Debugln("Stop wrapping new connections")
+				if !errors.As(parentContext.Err(), &context.Canceled) {
+					logger.WithError(parentContext.Err()).Debugln("Stop wrapping new connections")
+				}
 				return
 			}
 
-			wrappedConnection, clientID, err := server.config.ConnectionWrapper.WrapServer(context.TODO(), connection)
+			connectionContext := context.TODO()
```

Comment:
Do you mean that we need an ability to cancel actual connection in such a way? Does it relate to the previous usage of multiply listeners that create connections instead of one listener as currently?